File: elixir/04-control-flow/case_match.exs
defmodule TrafficLight do
  def action(color) do
    case color do
      :red -> "Stop"
      :yellow -> "Slow down"
      :green -> "Go"
      _ -> "Unknown signal"
    end
  end
end

IO.puts(TrafficLight.action(:red))
IO.puts(TrafficLight.action(:green))
IO.puts(TrafficLight.action(:blue))

File: erlang/04-control-flow/case_match.erl
#!/usr/bin/env escript

main(_) ->
    io:format("~s~n", [action(red)]),
    io:format("~s~n", [action(green)]),
    io:format("~s~n", [action(blue)]).

%% case picks the first matching pattern; _ is the catch-all
action(Color) ->
    case Color of
        red    -> "Stop";
        yellow -> "Slow down";
        green  -> "Go";
        _      -> "Unknown signal"
    end.
